torch/ao/quantization/pt2e/numeric_debugger.py

```diff
@@ -11,3 +20,188 @@ def generate_numeric_debug_handle(graph_module: GraphModule) -> None:
         if node.op != "placeholder" and NUMERIC_DEBUG_HANDLE_KEY not in node.meta:
             node.meta[NUMERIC_DEBUG_HANDLE_KEY] = unique_id
             unique_id += 1
+
+
+class OutputLogger(torch.nn.Module):
+    """
+    Base class for capturing output values.
+    """
```

Comment:
Can you add a better description? What types of output values does it capture? When would it not work?
I'd recommend focusing on the cases specific to prepare/convert for now